api/v1beta1/evaljob_types.go

```diff
@@ -0,0 +1,141 @@
+	NewJobState JobState = "New"
+	// The job is scheduled and waiting for available resources to run it
+	ScheduledJobState JobState = "Scheduled"
+	// The job is running
+	RunningJobState JobState = "Running"
+	// The job is complete
+	CompleteJobState JobState = "Complete"
+	// The job is cancelled
+	CancelledJobState JobState = "Cancelled"
+)
+
+// +kubebuilder:validation:Enum=NoReason;Succeeded;Failed;Cancelled
+type Reason string
+
+const (
+	// Job is still running and no final result yet
+	NoReason Reason = "NoReason"
+	// Job finished successfully
+	SucceedReason Reason = "Succeeded"
+	// Job failed
+	FailedReason Reason = "Failed"
+	// Job is cancelled
+	CancelledReason Reason = "Cancelled"
+)
+
+type Arg struct {
+	Name  string `json:"name"`
+	Value string `json:"value,omitempty"`
+}
+
+// EvalJobSpec defines the desired state of EvalJob
+type EvalJobSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Model name
+	Model string `json:"model"`
+	// Args for the model
+	// +optional
+	ModelArgs []Arg `json:"modelArgs,omitempty"`
+	// Evaluation tasks
+	Tasks []string `json:"tasks"`
+	// Sets the number of few-shot examples to place in context
+	// +optional
+	NumFewShot *int `json:"numFewShot,omitempty"`
+	// Accepts an integer, or a float between 0.0 and 1.0 . If passed, will limit
+	// the number of documents to evaluate to the first X documents (if an integer)
+	// per task or first X% of documents per task
+	// +optional
+	Limit string `json:"limit,omitempty"`
+	// Map to `--gen_kwargs` parameter for the underlying library.
+	// +optional
+	GenArgs []Arg `json:"genArgs,omitempty"`
+	// If this flag is passed, then the model's outputs, and the text fed into the
+	// model, will be saved at per-document granularity
+	// +optional
+	LogSamples *bool `json:"logSamples,omitempty"`
+}
+
+// EvalJobStatus defines the observed state of EvalJob
+type EvalJobStatus struct {
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// The name of the Pod that runs the evaluation job
+	// +optional
+	PodName string `json:"podName,omitempty"`
+	// State of the job
+	// +optional
+	State JobState `json:"state,omitempty"`
+	// Final result of the job
+	// +optional
+	Reason Reason `json:"reason,omitempty"`
+	// Message about the current/final status
+	// +optional
+	Message string `json:"message,omitempty"`
+	// Information when was the last time the job was successfully scheduled.
+	// +optional
+	LastScheduleTime *metav1.Time `json:"lastScheduleTime,omitempty"`
+	// Information when the job's state changes to Complete.
+	// +optional
+	CompleteTime *metav1.Time `json:"completeTime,omitempty"`
+	// Evaluation results
+	// +optional
+	Results string `json:"results,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// EvalJob is the Schema for the evaljobs API
+type EvalJob struct {
```

Comment:
I think we'll want to be careful about how we name the CR. I think `EvalJob` is too vague since there are numerous things that users may want to evaluate in a kube cluster. I think we could call it `LMEvalJob` to explicitly link it to the `lm-eval` python package.

Reply:
sure thing. This should be related to your other comment about the `GroupName` and `Kind`. The reason I have the `EvalJob` in the first place is that the GroupName is `lm-eval-service.github.com`.

If we change the GroupName to `foundation-model-stack.github.com`, then the `LMEvalJob` is definitely better.